build-system/tasks/check-sourcemaps.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const fs = require('fs');
+const log = require('fancy-log');
+const {cyan, green, red} = require('ansi-colors');
+const {execOrDie} = require('../common/exec');
+
+// Compile related constants
+const distWithSourcemapsCmd = 'gulp dist --core_runtime_only --full_sourcemaps';
+const v0JsMap = 'dist/v0.js.map';
+
+// Sourcemap URL related constants
+const sourcemapUrlMatcher =
+  'https://raw.githubusercontent.com/ampproject/amphtml/\\d{13}/';
+
+/**
+ * Throws an error with the given message
+ *
+ * @param {string} message
+ */
+function throwError(message) {
+  const err = new Error(message);
+  err.showStack = false;
+  throw err;
+}
+
+/**
+ * Build runtime with sourcemaps if needed.
+ */
+function maybeBuild() {
+  if (!argv.nobuild) {
+    log('Compiling', cyan('v0.js'), 'with full sourcemaps...');
+    execOrDie(distWithSourcemapsCmd, {'stdio': 'ignore'});
+  }
+}
+
+/**
+ * Verifies that a correctly formatted sourcemap URL is present in v0.js.map.
+ */
+function checkSourcemapUrl() {
+  log('Inspecting sourcemaps URL in', cyan(v0JsMap) + '...');
+  const v0JsMapJson = JSON.parse(fs.readFileSync(v0JsMap, 'utf8'));
+  if (!v0JsMapJson.sourceRoot) {
+    log(
+      red('ERROR:'),
+      'Could not find',
+      cyan('sourceRoot'),
+      'in',
+      cyan(v0JsMap)
+    );
+    throwError('Error in finding sourcemap URL');
+  }
+  if (!v0JsMapJson.sourceRoot.match(sourcemapUrlMatcher)) {
```

Comment:
Would this have caught one of the previous source map bugs? My read is "probably not" looking at the regex: `const sourcemapUrlMatcher = 'https://raw.githubusercontent.com/ampproject/amphtml/\\d{13}/'`.

The marginal CI cost appears to be non-zero, so I'd say let's ship if I'm wrong about the above or we can reduce the cost to zero. :)

Reply:
@jridgewell was the previous breakage in the URL format that's being tested here? Or was it something else?

Reply:
The previous breakage had `https://raw.githubusercontent.com/ampproject/amphtml/123/src/service/src/service/foo.js` (duplicate `src/service/src/service/`, which wouldn't be caught by this.

Reply:
Do either of you know which field contains that path? If that too is non-trivial, I'm okay with abandoning the PR based on this thread.

Reply:
`sources` array had these paths.

Reply:
Gotcha. Added a check to make sure all paths in the `sources` array are valid.